lib/widgets/compose_box.dart

```diff
@@ -158,8 +159,8 @@ class ComposeContentController extends ComposeController<ContentValidationError>
     value = value.replaced(
       replacementRange,
       url == null
-        ? '[Failed to upload file: $filename]()' // TODO(i18n)
-        : '[$filename](${url.toString()})');
+        ? inlineLink('Failed to upload file: $filename', null) // TODO(i18n)
```

Comment:
When I came back to this code, I wondered how helpful this "failure" content is. It'd be just as easy to delete the "loading" placeholder.

(Same for quote-and-reply.)

Reply:
(We'll already be showing an error dialog about the failure, I think.)

Reply:
Yeah, just deleting the placeholder makes sense to me.